runtime/python/pose_estimation/pose_estimation.py

```diff
@@ -0,0 +1,247 @@
+            processed_image, raw_detections, output_path, image_id,
+            height, width, class_num,
+            max_detections, score_threshold, nms_iou_thresh, regression_length, strides
+        )
+
+        image_id += 1
+
+
+def infer(
+    images: list[Image.Image],
+    net_path: str,
+    batch_size: int,
+    class_num: int,
+    output_path: Path,
+    data_type_dict: dict,
+    max_detections: int,
+    score_threshold: float,
+    nms_iou_thresh: float,
+    regression_length: int,
+    strides: list[int]
+) -> None:
+    """
+    Run inference with HailoAsyncInference, handle processes, and ensure proper cleanup.
+
+    Args:
+        images (list[Image.Image]): List of images to process.
+        net_path (str): Path to the HEF model file.
+        batch_size (int): Number of images per batch.
+        class_num (int): Number of classes.
+        output_path (Path): Path to save the output images.
+        data_type_dict (dict): Dictionary of layer names and data types.
+        max_detections (int): Maximal number of detections per class.
+        score_threshold (float): Confidence threshold for filtering.
+        nms_iou_thresh (float): IoU threshold for NMS.
+        regression_length (int): Maximum regression value for bounding boxes.
+        strides (list[int]): Stride values for each prediction scale.
+    """
+    input_queue = mp.Queue()
+    output_queue = mp.Queue()
+
+    hailo_inference = HailoAsyncInference(
+        net_path, input_queue, output_queue, batch_size, output_type=data_type_dict
+    )
+    height, width, _ = hailo_inference.get_input_shape()
+
+    enqueuing_process = Process(
+        target=enqueue_images,
+        args=(images, batch_size, input_queue, width, height)
+    )
+    processing_process = Process(
+        target=process_output,
+        args=(
+            output_queue, output_path, width, height, class_num, 
+            max_detections, score_threshold, nms_iou_thresh, regression_length, strides
+        )
+    )
+
+    enqueuing_process.start()
+    processing_process.start()
+
+    try:
+        hailo_inference.run()
+        enqueuing_process.join()
+        # To signal processing process to exit
+        output_queue.put(None)
+        processing_process.join()
+      
+        # Check for any exceptions in the processes
+        if enqueuing_process.exitcode or processing_process.exitcode != 0:
+            raise RuntimeError("Process terminated with an error")
+     
+        logging.info(f'Inference was successful! Results have been saved in {output_path}')
+      
+    except Exception as e:
+        logging.error(f"Inference error: {e}")
+
+        #TODO ask if needed if os._exit is done afterwards
+        # Ensure cleanup if there's an error
+        input_queue.close()
+        output_queue.close()
+        enqueuing_process.terminate()
+        processing_process.terminate()
+
+        os._exit(1)  # Force exit on error
```

Comment:
is this still necessary with the multiprocessing?

Reply:
yes...